File: languages/js/src/entity/identity_v2/network.ts
import { NetworkId as NetworkProto } from "../../bridge/proto/identity_entities_v2";

export enum NetworkId {
    UNRECOGNIZED = -1,
    MAIN = 0,
    MUMBAI = 1,
    GOERLI = 2,
    UNKNOWN_NETWORK = 3,
    NO_NETWORK = 4
}

export namespace NetworkId {
    export function toProto(network: NetworkId): NetworkProto {
        switch (network) {
            case NetworkId.MAIN:
                return NetworkProto.MAIN;
            case NetworkId.MUMBAI:
                return NetworkProto.MUMBAI;
            case NetworkId.GOERLI:
                return NetworkProto.GOERLI;
            case NetworkId.NO_NETWORK:
                return NetworkProto.NO_NETWORK
            case NetworkId.UNKNOWN_NETWORK:
                return NetworkProto.UNKNOWN_NETWORK
            default:
                return NetworkProto.UNRECOGNIZED;
        }
    }

    export function fromProto(network: NetworkProto | undefined): NetworkId {
        switch (network) {
            case NetworkProto.MAIN:
                return NetworkId.MAIN;
            case NetworkProto.MUMBAI:
                return NetworkId.MUMBAI;
            case NetworkProto.GOERLI:
                return NetworkId.GOERLI;
            case NetworkProto.NO_NETWORK:
                return NetworkId.NO_NETWORK;
            case NetworkProto.UNKNOWN_NETWORK:
                return NetworkId.UNKNOWN_NETWORK;
            default:
                return NetworkId.UNRECOGNIZED;
        }
    }
}

File: languages/python/bloock/entity/identity_v2/network.py
from __future__ import annotations

from enum import Enum
from bloock._bridge.proto.identity_entities_v2_pb2 import NetworkId as NetworkProto


class Network(Enum):
    MAIN = 0
    MUMBAI = 1
    GOERLI = 2
    NO_NETWORK = 3
    UNKNOWN_NETWORK = 4

    def __int__(self):
        return self.value

    @staticmethod
    def to_proto(network: Network) -> NetworkProto.ValueType:
        if network == Network.MAIN:
            return NetworkProto.MAIN
        elif network == Network.MUMBAI:
            return NetworkProto.MUMBAI
        elif network == Network.GOERLI:
            return NetworkProto.GOERLI
        elif network == Network.NO_NETWORK:
            return NetworkProto.NO_NETWORK
        elif network == Network.UNKNOWN_NETWORK:
            return NetworkProto.UNKNOWN_NETWORK
